ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/StandaloneApiServer.scala

```diff
@@ -211,3 +169,43 @@ class StandaloneIndexServer(
   }
 
 }
+
+object StandaloneApiServer {
+  private val actorSystemName = "index"
+
+  private val asyncTolerance = 30.seconds
+
+  private val sharedEngine = Engine()
+
+  private case class ApiServerState(
+      ledgerId: LedgerId,
+      apiServer: ApiServer,
+      indexAndWriteService: AutoCloseable,
+  ) extends AutoCloseable {
+    def port: Int = apiServer.port
+
+    override def close(): Unit = {
+      apiServer.close() //fully tear down the old server.
+      indexAndWriteService.close()
+    }
+  }
+
+  private case class Infrastructure(actorSystem: ActorSystem, materializer: ActorMaterializer)
+      extends AutoCloseable {
+    def executionContext: ExecutionContext = materializer.executionContext
+
+    override def close(): Unit = {
+      materializer.shutdown()
+      Await.result(actorSystem.terminate(), asyncTolerance)
+      ()
+    }
+  }
+
+  private case class SandboxState(apiServerState: ApiServerState, infra: Infrastructure)
```

Comment:
As noted about, I'd take advantage of the situation to re-evaluate if we really need this to be a `case class`. My rule of a thumb is that if it doesn't make sense for a class to be `Serializable`, it should not be a `case class`. :slightly_smiling_face:

Reply:
Done.